EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -374,6 +380,43 @@ int RawDataUnpacker::ProcessVFATDataParallel(const uint16_t *buf, unsigned int O
     presenceFlags |= 0x8;
     fd[0] = buf[dataOffset + 8];
   }
+  
+  // get channel data for diamond compact mode
+  if (hFlag == vmDiamondCompact)
+  {
+    for (unsigned int i = 1; (buf[i+1] & 0xFFF0)!= 0xF000; i++) {
+      if ( ( buf[i] & 0xF000 ) == VFAT_HEADER_OF_EC ) {
+        // Event Counter word is found
+        fd[10] = buf[i];
+        continue;
+      }
+      switch ( buf[i] & 0xF800 ) {
+        case VFAT_DIAMOND_HEADER_OF_WORD_2:
+          // word 2 of the diamond VFAT frame is found
+          fd[2] = buf[i];
+          fd[1] = buf[i + 1];
```

Comment:
just confirming that this is not a typo - fd[2] followed by fd[1] looks like one - might be worth a comment